### PR TITLE
Add user apps Kconfig for settings

### PR DESCRIPTION
**Pull request by louisom**
In this moment, build system will build all program in
folder user/apps, but this may cause unexpected effect,
for example, let kdb unusable in UART.

So, we can add a new kconfig file for apps like test cases
or real user program like pingpong, and thus rename menu
title from "Test Case" to "User Space".

And add submenu "Apps" and "Test Cases" for now.

The structure look like this:
- User Space
  - [ ] Compile with user apps
    - Apps
    - Test Cases

### CODE REVIEW

**Comment by grapherd** [comment]
It seems that travis CI broke by new settings....

```
$ build/host/conf --silentoldconfig build/host/Config.in
*
* Restart config...
*
*
* User Space
*
Compile with user apps (USER_APPS_INCLUDE) [N/y/?] (NEW) 
```

**Comment by jserv** [comment]
Cc. @georgekang

**Comment by jserv** [comment]
`user/apps` is not a folder. It is a "directory" instead.

File: user/Kconfig [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+# Copyright (c) 2015-2016 The F9 Microkernel Project. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-menu "Test Case"
-config EXTI_INTERRUPT_TEST
-	bool "EXTI interrupt test case"
+menu "User Space"
+
+config USER_APPS_INCLUDE
```
Comment (jserv) [inline user/Kconfig]: I dislike the use of `USER_APPS_INCLUDE`. The mnemonic alternative can be `USER_APPS_ENABLED`.

**Comment by jserv** [comment]
@grapherd, You have to update Travis-CI configurations right after new options are introduced.

File: user/Kconfig (hunk 1 above)
Comment (grapherd) [inline user/Kconfig]: That is true, I'll change it to `USER_APPS_ENABLED`

Comment (jserv) [inline user/Kconfig]: Hint: use `git rebase -i` to squash the changes.

**Comment by grapherd** [comment]
I found that there have still some problem need to be clear.

I can only success to see kdb when first `make defconfig` then disable `USER_APPS_ENABLED`.

If I first disable `USER_APPS_ENABLED` then `make flash`, it won't appear kdb, but only `==` on screen.

**Comment by jserv** [comment]
I expect there are only two commis: one is Kconfig changes, and another is Travis-CI specific update.

**Pull request merged**